modules/contact/include/userobjects/WeightedGapUserObject.h

```diff
@@ -36,6 +36,53 @@ class WeightedGapUserObject : public MortarUserObject
    */
   virtual const ADVariableValue & contactPressure() const = 0;
 
+  /**
+   * @param node Node pointer
+   * @return The normal contact pressure at the node
+   */
+  virtual Real getNormalContactPressure(const Node * const /*node*/) const
```

Comment:
`LMWeightedGapUserObject` doesn't override this one, so it'd become an abstract class.

Reply:
But it could, right? It would just return the LM nodal value?

Reply:
It could, I think. We have a Lagrange-multiplier formulation when we call `LMWeightedGapUserObject` as opposed to a penalty one. That is, the contact pressure values can be analyzed directly as variable outputs. The auxiliary kernels I am proposing in this PR are meant to leverage the penalty objects. Not that it cannot/should not be changed, but deeper changes would be required. As an example, we don't have accumulated slip in the frictional problem when enforced with Lagrange multipliers.

Let me know if you'd like me to extend that getter to the LM constraint.

Reply:
Nah it's fine